app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -672,6 +678,60 @@ fun CommentItem(
   }
 }
 
+@Composable
+fun PaymentInfoScreen(price: Double) {
+  var showDialog by remember { mutableStateOf(false) }
+
+  // Payment Section
+  Row(
+      verticalAlignment = Alignment.CenterVertically,
+      horizontalArrangement = Arrangement.Start,
+      modifier = Modifier.testTag("paymentSection")) {
+        // Payment Text
+        Text(
+            text = "Payment info",
+            style = MaterialTheme.typography.bodyMedium,
+            modifier = Modifier.padding(end = SMALL_PADDING.dp).testTag("paymentInfo"))
+
+        // Info Icon with Click
+        IconButton(modifier = Modifier.testTag("infoIconButton"), onClick = { showDialog = true }) {
+          Icon(
+              painter = painterResource(id = android.R.drawable.ic_dialog_info),
+              contentDescription = "Info",
+              tint = Color.Gray)
+        }
+      }
+
+  // Info Dialog
+  if (showDialog) {
+    AlertDialog(
+        modifier = Modifier.testTag("paymentInfoDialog"),
+        onDismissRequest = { showDialog = false },
+        confirmButton = {
+          TextButton(modifier = Modifier.testTag("okButton"), onClick = { showDialog = false }) {
+            Text(text = stringResource(id = R.string.ok))
+          }
+        },
+        title = {
+          Text(
+              modifier = Modifier.testTag("paymentInfoTitle"),
+              text = stringResource(id = R.string.payment_info))
+        },
+        text = {
+          if (price != 0.0) {
```

Comment:
I guess you assume no price could be negative. It may would have been better to consider this edge case

Reply:
You could for example show a message that says _"Invalid Price"_ when `price < 0.0`